### PR TITLE
new package xwmfs, see bug #634810

### PR DESCRIPTION
**Pull request by gerstner-hub**
I would like to add this package and would be willing to become proxy maintainer for it.

### CODE REVIEW

**Comment by jonasstein** [comment]
Why is this package important for the official tree? To me it sounds ideal for an overlay.

https://github.com/gerstner-hub/xwmfs writes:

> xwmfs is currently not broadly tested. Thus it is not necessarily very stable.

File: net-fs/xwmfs/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+  <maintainer type="person">
+    <email>matthias.gerstner@nefkom.net</email>
```
Comment (jonasstein) [inline net-fs/xwmfs/metadata.xml]: needs bugzilla account

File: net-fs/xwmfs/xwmfs-0.81.ebuild [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+DESCRIPTION="Display and control state of X11 windows and window manager in the file system"
+HOMEPAGE="https://github.com/gerstner-hub/xwmfs"
+SRC_URI="https://github.com/gerstner-hub/${PN}/releases/download/v${PV}/${P}-dist.tar.gz"
+
+LICENSE="GPL-2"
+KEYWORDS="amd64 ~arm ~hppa ~ppc ~ppc64 ~x86 ~x86-fbsd ~amd64-linux ~arm-linux ~x86-linux"
```
Comment (jonasstein) [inline net-fs/xwmfs/xwmfs-0.81.ebuild]: please have a look at "Keywording New Packages" https://devmanual.gentoo.org/keywording/

File: net-fs/xwmfs/xwmfs-0.81.ebuild [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+DESCRIPTION="Display and control state of X11 windows and window manager in the file system"
```
Comment (jonasstein) [inline net-fs/xwmfs/xwmfs-0.81.ebuild]: I do not understand this description.

**Comment by gerstner-hub** [comment]
Thanks for the review.

I adjusted the KEYWORDS and think ~amd64, ~x86 and ~arm would be justified since I tested the package on these platforms successfully.

File: net-fs/xwmfs/metadata.xml (hunk 1 above)
Comment (gerstner-hub) [reply net-fs/xwmfs/metadata.xml]: I've synced mail addresses of my bugzilla account and the `metadata.xml`. It should match now.

**Comment by gerstner-hub** [comment]
> Please check, what happens with your manpage, DOCS, examples.

Do you think there is something wrong? It looks good to me:

```
$ equery f xwmfs
[...]
/usr/share/doc/xwmfs-0.81/AUTHORS
/usr/share/doc/xwmfs-0.81/ChangeLog.bz2
/usr/share/doc/xwmfs-0.81/NEWS.bz2
/usr/share/doc/xwmfs-0.81/README.bz2
/usr/share/man
/usr/share/man/man1
/usr/share/man/man1/xwmfs.1.bz2
```

**Comment by gerstner-hub** [comment]
> Why is this package important for the official tree? To me it sounds ideal for an overlay.

I wasn't aware a package needs to qualify as important for the official tree. What makes a package important?

Let's say the package would add to the liveliness and versatility of Gentoo :-)

I've always found it great that Gentoo also covers less popular or niche packages.

Which overlay would you suggest? What would be the process of getting a package from the overlay into the official tree, or is that not intended at all?
 
> https://github.com/gerstner-hub/xwmfs writes:
> 
> xwmfs is currently not broadly tested. Thus it is not necessarily very stable.

Well since that sentence came from myself: I'm a conservative person and don't want to raise wrong expectations. Still there are no known bugs in the package and the overall quality is good I'd say.

File: net-fs/xwmfs/metadata.xml (hunk 1 above)
Comment (jonasstein) [reply net-fs/xwmfs/metadata.xml]: fine

**Comment by jonasstein** [comment]
I would suggest betagarden as overlay. This overlay was set up for packages which are not often used or need more testing. If it proves to run stable and several users will use it, we can merge it to the official tree. Would this be fine for you?

**Comment by gerstner-hub** [comment]
Okay, betagarden sounds good. Do I need to do anything else for this?

**Comment by jonasstein** [comment]
OK I discussed this with another dev too and he agreed with me. I will merge it to betagarden for you.
But I am not satisfied with the category yet. Please read `/usr/portage/net-fs/metadata.xml` 
I think that is something different. Please suggest a better category.

**Comment by gerstner-hub** [comment]
> But I am not satisfied with the category yet. Please read /usr/portage/net-fs/metadata.xml
> I think that is something different. Please suggest a better category.

Indeed. I took this over from sshfs and overlooked it. I've changed it to the sys-fs category.

**Pull request closed without merging**